PropertySheets/Platform.props

```diff
@@ -1,228 +1,241 @@
-      <ErrorCheckBounds>false</ErrorCheckBounds>
-      <ErrorCheckEnumRange>false</ErrorCheckEnumRange>
-      <ErrorCheckRefPointers>false</ErrorCheckRefPointers>
-      <ErrorCheckStubData>false</ErrorCheckStubData>
-      <ValidateAllParameters>true</ValidateAllParameters>
-      <StructMemberAlignment>NotSet</StructMemberAlignment>
-    </Midl>
-    <ResourceCompile>
-      <Culture>0x0000</Culture>
-      <IgnoreStandardIncludePath>false</IgnoreStandardIncludePath>
-      <ShowProgress>false</ShowProgress>
-      <ResourceOutputFileName>$(IntDir)%(Filename).res</ResourceOutputFileName>
-    </ResourceCompile>
-    <Manifest>
-      <OutputManifestFile>$(IntDir)$(TargetFileName).embed.manifest</OutputManifestFile>
-      <SuppressStartupBanner>true</SuppressStartupBanner>
-      <VerboseOutput>false</VerboseOutput>
-      <GenerateCatalogFiles>false</GenerateCatalogFiles>
-      <UpdateFileHashes>false</UpdateFileHashes>
-    </Manifest>
-    <ManifestResourceCompile>
-      <ResourceOutputFileName>$(IntDir)$(TargetFileName).embed.manifest.res</ResourceOutputFileName>
-    </ManifestResourceCompile>
-    <Xdcmake>
-      <DocumentLibraryDependencies>true</DocumentLibraryDependencies>
-      <SuppressStartupBanner>true</SuppressStartupBanner>
-      <OutputFile>$(TargetDir)$(TargetName).xml</OutputFile>
-    </Xdcmake>
-  </ItemDefinitionGroup>
-  <!-- END common platform settings -->
-
-  <!-- Win32 -->
-  <ItemDefinitionGroup Condition="'$(Platform)'=='Win32'">
-    <Lib>
-      <TargetMachine>MachineX86</TargetMachine>
-    </Lib>
-    <Link>
-      <TargetMachine>MachineX86</TargetMachine>
-    </Link>
-  </ItemDefinitionGroup>
-  <!-- END Win32 -->
-
-  <!-- x64 -->
-  <PropertyGroup Condition="'$(Platform)'=='x64'">
-    <OutDir>$(SolutionDir)bin\$(Configuration)64\</OutDir>
-    <IntDir>$(SolutionDir)build\$(Configuration)64\$(ProjectName)\</IntDir>
-  </PropertyGroup>
-
-  <PropertyGroup Condition="'$(ConfigurationType)|$(Platform)'=='StaticLibrary|x64'">
-    <OutDir>$(SolutionDir)bin\$(Configuration)64\lib\</OutDir>
-  </PropertyGroup>
-
-  <ItemDefinitionGroup Condition="'$(Platform)'=='x64'">
-    <Lib>
-      <TargetMachine>MachineX64</TargetMachine>
-    </Lib>
-    <Link>
-      <TargetMachine>MachineX64</TargetMachine>
-    </Link>
-  </ItemDefinitionGroup>
-  <!-- END x64 -->
-
-</Project>
\ No newline at end of file
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project DefaultTargets="Build" ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup Label="Globals">
+    <WindowsTargetPlatformVersion Condition="'$(WindowsTargetPlatformVersion)' == '' AND '$(PlatformToolset)' == 'v142'">10.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion Condition="'$(WindowsTargetPlatformVersion)' == '' AND '$(VisualStudioVersion)' >= 16.0">10.0.19041.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformMinVersion>8.1</WindowsTargetPlatformMinVersion>
+  </PropertyGroup>
```

Comment:
VS2015 can also use SDK 10.0, but it is not mandatory to set up an SDK to start with. It's only mandatory to establish the SDK from VS2017 onwards.

These implementations are more robust:
https://github.com/chakra-core/ChakraCore/blob/master/Build/Chakra.Build.Default.props#L5-L14
https://github.com/gonetz/GLideN64/blob/master/projects/msvc/GLideN64.vcxproj#L57-L60

Fixed versions like `10.0.19041.0` are not guaranteed in Enterprise editions for stability, `10.0` is safe.

Reply:
Thanks for the reference.
I'm aware I was sort of hard coding the specific 10.0.X version, which should last for a while, but Gonetz's approach looks better.

I'll update this.